astro-client-iam-core/client.go

```diff
@@ -0,0 +1,66 @@
+package astroiamcore
+
+import (
+	"bytes"
+	httpContext "context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"runtime"
+
+	"github.com/astronomer/astro-cli/context"
+	"github.com/astronomer/astro-cli/pkg/httputil"
+	"github.com/astronomer/astro-cli/version"
+)
+
+var (
+	ErrorRequest  = errors.New("failed to perform request")
+	ErrorBaseURL  = errors.New("invalid baseurl")
+	HTTPStatus200 = 200
+	HTTPStatus204 = 204
+)
+
+// a shorter alias
+type CoreClient = ClientWithResponsesInterface
+
+func requestEditor(ctx httpContext.Context, req *http.Request) error {
+	currentCtx, err := context.GetCurrentContext()
+	if err != nil {
+		return nil
+	}
+	os := runtime.GOOS
+	arch := runtime.GOARCH
+	baseURL := currentCtx.GetPublicRESTAPIURL("iam/v1beta1")
+	requestURL, err := url.Parse(baseURL + req.URL.String())
+	if err != nil {
+		return fmt.Errorf("%w, %s", ErrorBaseURL, baseURL)
+	}
+	req.URL = requestURL
+	req.Header.Add("authorization", currentCtx.Token)
+	req.Header.Add("x-astro-client-identifier", "cli")
+	req.Header.Add("x-astro-client-version", version.CurrVersion)
+	req.Header.Add("x-client-os-identifier", os+"-"+arch)
+	req.Header.Add("User-Agent", fmt.Sprintf("astro-cli/%s", version.CurrVersion))
+	return nil
+}
+
+// create api client for astro core services
+func NewCoreClient(c *httputil.HTTPClient) *ClientWithResponses {
```

Comment:
```suggestion
func NewIamCoreClient(c *httputil.HTTPClient) *ClientWithResponses {
```

Reply:
Can we rename this to `NewIAMCoreClient`?